tfhe/src/integer/gpu/server_key/radix/scalar_comparison.rs

```diff
@@ -120,6 +120,144 @@ impl CudaServerKey {
         result
     }
 
+    /// # Safety
+    ///
+    /// - `stream` __must__ be synchronized to guarantee computation has finished, and inputs must
+    ///   not be dropped until stream is synchronised
+    pub unsafe fn unchecked_scalar_eq_async<T>(
+        &self,
+        ct: &CudaRadixCiphertext,
+        scalar: T,
+        stream: &CudaStream,
+    ) -> CudaRadixCiphertext
+    where
+        T: DecomposableInto<u64>,
+    {
+        self.unchecked_scalar_comparison_async(ct, scalar, ComparisonType::EQ, stream)
+    }
+
+    pub fn unchecked_scalar_eq<T>(
+        &self,
+        ct: &CudaRadixCiphertext,
+        scalar: T,
+        stream: &CudaStream,
+    ) -> CudaRadixCiphertext
+    where
+        T: DecomposableInto<u64>,
+    {
+        let result = unsafe { self.unchecked_scalar_eq_async(ct, scalar, stream) };
+        stream.synchronize();
+        result
+    }
+
+    /// # Safety
+    ///
+    /// - `stream` __must__ be synchronized to guarantee computation has finished, and inputs must
+    ///   not be dropped until stream is synchronised
+    pub unsafe fn scalar_eq_async<T>(
+        &self,
+        ct: &CudaRadixCiphertext,
+        scalar: T,
+        stream: &CudaStream,
+    ) -> CudaRadixCiphertext
+    where
+        T: DecomposableInto<u64>,
+    {
+        let mut tmp_lhs;
+        let lhs = if ct.block_carries_are_empty() {
+            ct
+        } else {
+            tmp_lhs = ct.duplicate_async(stream);
+            self.full_propagate_assign_async(&mut tmp_lhs, stream);
+            &tmp_lhs
+        };
+
+        self.unchecked_scalar_eq_async(lhs, scalar, stream)
+    }
+
+    pub fn scalar_eq<T>(
```

Comment:
@pdroalves could you add doc tests for all default operations like this one please? :slightly_smiling_face:

Reply:
Sure! It is done. I had to write my own doc test since there is none for the CPU, and this made me realize that most of the comparison operations don't have too. Maybe we should open a new PR to add all missing doc tests.

Reply:
I've opened an issue -> https://github.com/zama-ai/tfhe-rs-internal/issues/454

Reply:
Thank you very much :slightly_smiling_face: